### PR TITLE
Replace JSCS with ESLint

### PR DESCRIPTION
**Pull request by Turbo87**
This PR replaces the few existing JSCS tests with ESLint rules (similar to #5455)

/cc @kellyselden @rwjblue

### CODE REVIEW

File: tests/unit/lint-test.js [hunk 1]
```diff
@@ -8,13 +8,8 @@ var paths = glob('tests/*').filter(function(path) {
...
   'lib',
-  'bin'
-]);
-
-require('mocha-jscs')(paths);
-
-paths = paths.concat([
-  'blueprints'
+  'bin',
+  'blueprints',
```
Comment (kellyselden) [inline tests/unit/lint-test.js]: trailing comma

Comment (Turbo87) [inline tests/unit/lint-test.js]: we don't want trailing commas?

Comment (kellyselden) [inline tests/unit/lint-test.js]: just not on the last one

Comment (Turbo87) [inline tests/unit/lint-test.js]: because IE8 compat?! :laughing:

Comment (Turbo87) [inline tests/unit/lint-test.js]: @kellyselden removed the comma and rebased on `master`

Comment (kellyselden) [inline tests/unit/lint-test.js]: Thanks.

**Comment by kellyselden** [comment]
I'm ok with this. JSCS was shortlived :cry: We hardly knew ye.

**Comment by Turbo87** [comment]
@kellyselden sorry :wink:

**Pull request merged**